### DISCUSSION TITLE
How to abort a request in server/api

### DISCUSSION THREAD

**Discussion by Gonerman** [nuxt/nuxt#26025] [Questions]
I have a server/api where I access the backend and give the response to the components. I need to abort the request for example onunmounted. In components the request is interrupted, but server/api continues to work. How can I make server/api to interrupt its request?

**Comment by Dino-Kupinic** [comment]
Hey there!
I think you can use the native `AbortController()`
https://developer.mozilla.org/en-US/docs/Web/API/AbortController

Here is a simple example:

```ts
const controller = new AbortController()
const timeoutId = setTimeout(() => controller.abort(), 5000)

const response = await fetch("URL", {signal: controller.signal})
clearTimeout(timeoutId)
```

We abort the request if we don't get a response within 5 seconds.
